passes/pmgen/split_shiftx.pmg

```diff
@@ -0,0 +1,59 @@
+state <SigSpec> shiftxB
+
+match shiftx
+	select shiftx->type == $shiftx
+	select param(shiftx, \Y_WIDTH).as_int() > 1
+endmatch
+
+match macc
+	select macc->type == $macc
+	select param(macc, \B_WIDTH).as_int() == 0
+	optional
+endmatch
+
+code shiftxB
+	shiftxB = port(shiftx, \B);
+
+	if (macc) {
+		const int b_width = param(shiftx, \B_WIDTH).as_int();
+		if (param(shiftx, \B_SIGNED) != 0 && shiftxB[b_width-1] == RTLIL::S0)
+			shiftxB = shiftxB.extract(0, b_width-1);
```

Comment:
None of this depends on macc so it should probably be done right after matching shiftx.

Reply:
See above re: trimming constant-0 sign bit.